drivers/jdbc-driver/src/main/kotlin/app/cash/sqldelight/driver/jdbc/JdbcDriver.kt

```diff
@@ -261,20 +253,28 @@ class JdbcPreparedStatement(
     }
   }
 
-  fun bindObjectOther(index: Int, obj: Any?) {
+  fun bindObject(index: Int, obj: Any?, type: Int) {
```

Comment:
I think this makes this a binary-incompatible change, right?

Reply:
Yes, we can also keep it but it's only used by the Postgres driver. 
And according to Alec  only the runtime needs to be api compatible.

Reply:
I think that was pre-2.0.

Reply:
Binary compatibility matters less for a driver like R2DBC which is still more "experimental", but the JDBC driver is widely used (and the SQLite one is an implementation of it) so we should definitely maintain compatibility for it.

We should enable the API dumper plugin!

Reply:
💁 I can refactor uses of `bindObjectOther` in the Postgres dialect to use this new api once it has been merged, then remove `bindObjectOther` before any new release.
It's only recently been added in the snapshot builds - so can be changed.

Reply:
> I can refactor uses of bindObjectOther

This is already part of this PR, but thanks.

> It's only recently been added in the snapshot builds - so can be changed.

In this case I agree, we can change it.

> We should enable the API dumper plugin!

Yes, will do it in another PR.